### PR TITLE
Add Enclave block and Batch health checks

### PR DESCRIPTION
**Pull request by otherview**
### Why this change is needed

https://github.com/ten-protocol/ten-internal/issues/2631

- Adds health check for l1 block ( last processed )
- Adds health check for l2 batch ( last executed)
- Adds async.Timestamp to avoid concurrent accesses to the timestamp

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/enclave.go [hunk 1]
```diff
@@ -1248,9 +1248,23 @@ func (e *enclaveImpl) HealthCheck() (bool, common.SystemError) {
 		e.logger.Info("HealthCheck failed for the enclave storage", log.ErrKey, err)
 		return false, nil
 	}
+
 	// todo (#1148) - enclave healthcheck operations
```
Comment (zkokelj) [inline go/enclave/enclave.go:1252]: Is this todo now done and should be removed?

File: go/enclave/storage/storage.go [hunk 2]
```diff
@@ -266,10 +266,14 @@ func (s *storageImpl) HealthCheck() (bool, error) {
 	defer s.logDuration("HealthCheck", measure.NewStopwatch())
 	headBatch, err := s.FetchHeadBatch()
 	if err != nil {
-		s.logger.Info("HealthCheck failed for enclave storage", log.ErrKey, err)
```
Comment (zkokelj) [inline go/enclave/storage/storage.go:269]: Why was this log removed?

File: go/enclave/components/block_processor.go [hunk 3]
```diff
@@ -77,9 +82,19 @@ func (bp *l1BlockProcessor) Process(br *common.BlockAndReceipts) (*BlockIngestio
 
 	h := br.Block.Hash()
 	bp.currentL1Head = &h
+	bp.lastIngestedBlock.Mark()
 	return ingestion, nil
 }
 
+func (bp *l1BlockProcessor) HealthCheck() (bool, error) {
```
Comment (zkokelj) [inline go/enclave/components/block_processor.go:89]: Maybe add a comment to describe what we consider healthy / unhealthy.

**Review by zkokelj — APPROVED** [review]
Nice work 👏 

Minor comments added (mostly about comments), but nothing blocking / major.

**Review by StefanIliev545 — APPROVED** [review]
LGTM

File: go/enclave/enclave.go (hunk 1 above)
Comment (otherview) [reply go/enclave/enclave.go:1252]: I'd leave is as I think there might be more areas to health check in the enclave.

File: go/enclave/storage/storage.go (hunk 2 above)
Comment (otherview) [reply go/enclave/storage/storage.go:269]: It's a duplicate log, it logs the error downstream

File: go/enclave/components/block_processor.go (hunk 3 above)
Comment (otherview) [reply go/enclave/components/block_processor.go:89]: Good idea 👍

**Pull request merged**